### PR TITLE
More robust retry mechanism for downloading rpms

### PR DESCRIPTION
**Pull request by tkopecek**
Can fix a few other cases of failures.

### CODE REVIEW

**Comment by tkopecek** [comment]
@praiskup

File: mock/py/mock-hermetic-repo.py [hunk 1]
```diff
@@ -16,17 +16,29 @@
...
-import backoff
 import requests
 
 logging.basicConfig(level=logging.DEBUG)
 log = logging.getLogger(__name__)
 
 
-@backoff.on_exception(backoff.expo, requests.exceptions.RequestException,
-                      max_tries=5, max_time=300)
+def request_with_retry(retries=5, backoff_factor=0.3,
```
[the remark this answers is not in the document]
Comment (praiskup) [reply mock/py/mock-hermetic-repo.py:29]: Can you wrap, perhaps into a docstring, reference to koji so we keep in sync?
https://pagure.io/koji/blob/fccf4fa3f990ca454a411c8a699e693f4e5b0ac8/f/koji/__init__.py#_2004

**Comment by praiskup** [comment]
Nit: pylint + backref to Koji code, but otherwise LGTM (maybe you want to rebase on top of current main to make CI green).

**Comment by tkopecek** [comment]
updated

**Review by praiskup — APPROVED** [review]
LGTM

**Pull request merged**